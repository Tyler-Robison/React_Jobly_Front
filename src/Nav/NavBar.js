import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import './NavBar.css'
import UserContext from "../context/UserContext";
import "bootstrap/dist/css/bootstrap.min.css"
import { Navbar, Container } from "react-bootstrap";
import { Button } from "@material-ui/core";

const NavBar = ({ logout }) => {

    const { currentUser } = useContext(UserContext)

    const loggedInView = <>

        <NavLink className="nav-link" to="/companies">
            Companies
        </NavLink>


        <NavLink className="nav-link" to="/jobs">
            Jobs
        </NavLink>

        <NavLink className="nav-link" to="/profile">
            Profile
        </NavLink>

        {currentUser ?
            <Button onClick={logout} variant="contained" color="primary"> Log Out {currentUser.firstName}</Button> :
            <Button onClick={logout}> Log Out   </Button>}

    </>

    const loggedOutView = <>

        <NavLink className="nav-link" to="/signup">
            Signup
        </NavLink>

        <NavLink className="nav-link" to="/login">
            Login
        </NavLink>

    </>

    return (
        <Navbar bg="light" expand="lg">
            <Container>
                <Navbar.Brand href="jobly">Jobly</Navbar.Brand>

                {currentUser ? loggedInView : loggedOutView}

            </Container>
        </Navbar>
    )
}

export default NavBar;
