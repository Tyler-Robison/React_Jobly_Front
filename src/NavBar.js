import React, { useContext } from "react";
import { NavLink } from "react-router-dom";
import './Styles/NavBar.css'
import UserContext from "./context/UserContext";

const NavBar = ({ logout, token }) => {

    const currentUser = useContext(UserContext)


    const loggedInView = <>
        <li>
            <NavLink className="nav-link" to="/profile">
                Profile
            </NavLink>
        </li>
        <li>
            {currentUser ?
                <button onClick={logout}> Log Out {currentUser.firstName}   </button> :
                <button onClick={logout}> Log Out   </button>}
        </li>
    </>

    const loggedOutView = <>
        <li>
            <NavLink className="nav-link" to="/signup">
                Signup
            </NavLink>
        </li>
        <li>
            <NavLink className="nav-link" to="/login">
                Login
            </NavLink>
        </li>
    </>

    return (
        <div>
            <ul>
                <li>
                    <NavLink className="nav-link" to="/jobly">
                        Jobly
                    </NavLink>
                </li>
                <li>
                    <NavLink className="nav-link" to="/companies">
                        Companies
                    </NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/jobs">
                        Jobs
                    </NavLink>
                </li>
                {currentUser ? loggedInView : loggedOutView}

            </ul>
        </div>
    )
}

export default NavBar;