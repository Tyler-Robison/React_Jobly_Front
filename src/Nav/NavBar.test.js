import { render, screen } from '@testing-library/react';
import NavBar from './NavBar';
import { MemoryRouter } from "react-router-dom";
import UserProvider from '../testUtils'

it("renders without crashing", function () {
    <MemoryRouter>
        <NavBar />
    </MemoryRouter>
});

it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
    const container = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <NavBar />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(container.asFragment()).toMatchSnapshot();
});