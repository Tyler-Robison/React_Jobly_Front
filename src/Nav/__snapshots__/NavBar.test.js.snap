// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="jobly"
        >
          Jobly
        </a>
        <a
          class="nav-link"
          href="/companies"
        >
          Companies
        </a>
        <a
          class="nav-link"
          href="/jobs"
        >
          Jobs
        </a>
        <a
          class="nav-link"
          href="/profile"
        >
          Profile
        </a>
        <button>
           Log Out Janet   
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="jobly"
        >
          Jobly
        </a>
        <a
          class="nav-link"
          href="/signup"
        >
          Signup
        </a>
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
